plugins/LayerView/LayerView.qml

```diff
@@ -214,72 +218,47 @@ Item
-            property bool extruder1_checked: UM.Preferences.getValue("layerview/extruder1_opacity") > 0.5
-            property bool extruder2_checked: UM.Preferences.getValue("layerview/extruder2_opacity") > 0.5
-            property bool extruder3_checked: UM.Preferences.getValue("layerview/extruder3_opacity") > 0.5
+            property var extruder_opacities: UM.Preferences.getValue("layerview/extruder_opacities").split(",")
             property bool show_travel_moves: UM.Preferences.getValue("layerview/show_travel_moves")
             property bool show_support: UM.Preferences.getValue("layerview/show_support")
             property bool show_adhesion: UM.Preferences.getValue("layerview/show_adhesion")
             property bool show_skin: UM.Preferences.getValue("layerview/show_skin")
             property bool show_infill: UM.Preferences.getValue("layerview/show_infill")
+            property bool only_show_top_layers: UM.Preferences.getValue("view/only_show_top_layers")
+            property int top_layer_count: UM.Preferences.getValue("view/only_show_top_layers")
 
             anchors.top: UM.LayerView.compatibilityMode ? compatibilityModeLabel.bottom : layerTypeCombobox.bottom
             anchors.topMargin: UM.Theme.getSize("default_margin").height
             anchors.left: parent.left
             anchors.leftMargin: UM.Theme.getSize("default_margin").width
 
-            CheckBox {
-                checked: view_settings.extruder0_checked
-                onClicked: {
-                    UM.Preferences.setValue("layerview/extruder0_opacity", checked ? 1.0 : 0.0);
-                }
-                text: "Extruder 1"
-                visible: !UM.LayerView.compatibilityMode && (UM.LayerView.extruderCount >= 1)
-            }
-            CheckBox {
-                checked: view_settings.extruder1_checked
-                onClicked: {
-                    UM.Preferences.setValue("layerview/extruder1_opacity", checked ? 1.0 : 0.0);
-                }
-                text: "Extruder 2"
-                visible: !UM.LayerView.compatibilityMode && (UM.LayerView.extruderCount >= 2)
-            }
-            CheckBox {
-                checked: view_settings.extruder2_checked
-                onClicked: {
-                    UM.Preferences.setValue("layerview/extruder2_opacity", checked ? 1.0 : 0.0);
-                }
-                text: "Extruder 3"
-                visible: !UM.LayerView.compatibilityMode && (UM.LayerView.etruderCount >= 3)
-            }
-            CheckBox {
-                checked: view_settings.extruder3_checked
-                onClicked: {
-                    UM.Preferences.setValue("layerview/extruder3_opacity", checked ? 1.0 : 0.0);
+            Repeater {
+                model: UM.LayerView.extruderCount
+                CheckBox {
+                    checked: [undefined, ""].indexOf(view_settings.extruder_opacities[index]) >= 0 || view_settings.extruder_opacities[index] > 0.5
```

Comment:
"Pythonic" means a combination of "elegant" and "using correct language syntax for the job". As far as I'm concerned these two attributes can be extended to other languages as well. If it's good for Python it should be good for Javascript, as long as this is also the most elegant syntax for the job.